src/sentry/rules/history/preview.py

```diff
@@ -62,13 +70,27 @@ def preview(
     Returns groups that would have triggered the given conditions and filters in the past 2 weeks
     """
     issue_state_conditions, frequency_conditions = categorize_conditions(conditions)
-
     # must have at least one condition to filter activity
     if not issue_state_conditions and not frequency_conditions:
         return None
-    # all the issue state conditions are mutually exclusive
     elif len(issue_state_conditions) > 1 and condition_match == "all":
-        return {}
+        # Of the supported conditions, any more than two would be mutually exclusive
+        if len(issue_state_conditions) > 2:
+            return {}
+
+        condition_ids = {condition["id"] for condition in issue_state_conditions}
+
+        # all the issue state conditions are mutually exclusive
+        if not any(condition in VALID_CONDITION_PAIRS for condition in condition_ids):
+            return {}
+
+        # if there are multiple issue state conditions, they must be one of the valid pairs
+        for condition in condition_ids:
+            if (
+                condition in VALID_CONDITION_PAIRS
+                and VALID_CONDITION_PAIRS[condition] not in condition_ids
```

Comment:
for `VALID_CONDITION_PAIRS`, is there a guaranteed sorting? do you need the reverse in the dictionary as well?

Reply:
No guaranteed order and I think this would've worked since we're iterating through the entire list of conditions, but added the reverse to make it explicit and clearer.